pkg/prometheus/statefulset.go

```diff
@@ -1076,3 +676,488 @@ func queryLogFilePath(queryLogFile string) string {
 
 	return filepath.Join(defaultQueryLogDirectory, queryLogFile)
 }
+
+// buildCommonPrometheusArgs builds an slice of arguments that are common between Prometheus Server and Agent.
+func buildCommonPrometheusArgs(cpf monitoringv1.CommonPrometheusFields, cg *ConfigGenerator, webRoutePrefix string) []monitoringv1.Argument {
+	promArgs := []monitoringv1.Argument{
+		{Name: "web.console.templates", Value: "/etc/prometheus/consoles"},
+		{Name: "web.console.libraries", Value: "/etc/prometheus/console_libraries"},
+		{Name: "config.file", Value: path.Join(confOutDir, configEnvsubstFilename)},
+		{Name: "web.enable-lifecycle"},
+	}
+
+	if cpf.Web != nil {
+		if cpf.Web.PageTitle != nil {
+			promArgs = cg.WithMinimumVersion("2.6.0").AppendCommandlineArgument(promArgs, monitoringv1.Argument{Name: "web.page-title", Value: *cpf.Web.PageTitle})
+		}
+
+		if cpf.Web.MaxConnections != nil {
+			promArgs = append(promArgs, monitoringv1.Argument{Name: "web.max-connections", Value: fmt.Sprintf("%d", *cpf.Web.MaxConnections)})
+		}
+	}
+
+	if cpf.EnableAdminAPI {
+		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.enable-admin-api"})
+	}
+
+	if cpf.EnableRemoteWriteReceiver {
```

Comment:
Is agent mode really okay with this argument? 🤔

Reply:
That's a very interesting question, my guess is that it shouldn't work 🤔

Reply:
Indeed it works as simon described!

I've made the following experiment:

Started a prometheus server with the following command line `./prometheus --config.file=documentation/examples/prometheus.yml --web.enable-remote-write-receiver`, and the config file below:

<details><summary>Prometheus server config</summary>

```yaml
global:
  scrape_interval: 15s
  evaluation_interval: 15s 

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9090"]
```

</details>


Next I started an agent, with remote-write receiver enabled `./prometheus --web.listen-address="0.0.0.0:9091" --enable-feature=remote-write-receiver,agent --config.file ../../documentation/examples/prometheus-agent.yml`

<details><summary>Prometheus agent1 config</summary>

```yaml
global:
  scrape_interval: 15s 
  evaluation_interval: 15s
  external_labels:
    prometheus_relay: true

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9091"]

remote_write:
  - url: "http://localhost:9090/api/v1/write"
```

</details>

Finally a second agent, with the following command `./prometheus --web.listen-address="0.0.0.0:9092" --enable-feature=agent --config.file ../../documentation/examples/prometheus-agent2.yml --storage.agent.path=data-agent2/`

<details><summary>Prometheus agent2 config</summary>

```yaml
global:
  scrape_interval: 15s 
  evaluation_interval: 15s
  external_labels:
    prometheus_relay: true

scrape_configs:
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9092"]

remote_write:
  - url: "http://localhost:9091/api/v1/write"
```

</details>

When querying the Prometheus server, I can see metrics with both agents' external labels, which proves that the first agent worked as a proxy between the server and the second agent 🙂 
![image](https://user-images.githubusercontent.com/24193764/219792048-3b409a8f-7965-4617-b0a2-905d7220e800.png)